File: packages/meta-pass-dump.meta/scripts/sanitize-dump.mjs
import { readFileSync, writeFileSync } from "node:fs"
import { relative, dirname } from "node:path"
import { fileURLToPath } from "node:url"

const scriptDir = dirname(fileURLToPath(import.meta.url))
const pkgDir = dirname(scriptDir)
const projectRoot = dirname(dirname(pkgDir))

for (const filepath of process.argv.slice(2)) {
  let content = readFileSync(filepath, "utf-8")
  content = content.replace(/"(\/[^"]*)"/g, (_, path) => {
    if (!path.startsWith(projectRoot)) return `"${path}"`
    const rel = relative(pkgDir, path)
    return `"${rel}"`
  })
  writeFileSync(filepath, content)
}

File: packages/meta-pass-dump.meta/scripts/sanitize-dump.py
import sys
import os
import re

script_dir = os.path.dirname(os.path.realpath(sys.argv[0]))
pkg_dir = os.path.dirname(script_dir)

def replace(m):
    path = m.group(1)
    rel = os.path.relpath(path, pkg_dir)
    return f'"{rel}"'

for filepath in sys.argv[1:]:
    with open(filepath, 'r') as f:
        content = f.read()
    content = re.sub(r'"(/[^"]*)"', replace, content)
    with open(filepath, 'w') as f:
        f.write(content)
